### PR TITLE
docs: add DetectIntent samples 

### PR DESCRIPTION
**Pull request by aribray**
Adds the following DetectIntent samples:

- detectIntent with IntentInput
- detectIntent with EventInput
- detectIntent with synthesized TTS output
- detectIntent with disabled webhook
- detectIntent with sentiment analysis
- streamingDetectIntent with partial response enabled
- detectIntent with AudioInput

### CODE REVIEW

**Review by kweinmeister — APPROVED** [review]

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  // DialogFlow API Detect Intent sample with Audio input.
+  public static void main(String[] args) throws IOException, ApiException {
+    String projectId = "my-project-id";
+    String locationId = "my-location-id";
+    String agentId = "my-agent-id";
+    String audioFileName = "my-audio-file-name";
+    int sampleRateHertz = 16000;
+    String sessionId = "my-session-id";
+    String languageCode = "my-language-code";
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java:46]: A lot of stuff to customize.  I wonder if you might be better off defaulting some of this stuff a commenting.

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import java.io.FileInputStream;
+import java.io.IOException;
+
+public class DetectIntentAudioInput {
+
+  // DialogFlow API Detect Intent sample with Audio input.
+  public static void main(String[] args) throws IOException, ApiException {
+    String projectId = "my-project-id";
+    String locationId = "my-location-id";
+    String agentId = "my-agent-id";
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java:42]: If this should really be a UUID from L73 below, then perhaps you should show that.

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    String audioFileName = "my-audio-file-name";
+    int sampleRateHertz = 16000;
+    String sessionId = "my-session-id";
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java:45]: L73 says this is a UUID, perhaps we should show that?

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      sessionsSettingsBuilder.setEndpoint(locationId + "-dialogflow.googleapis.com:443");
+    }
+    SessionsSettings sessionsSettings = sessionsSettingsBuilder.build();
+
+    // Instantiates a client
+    try (SessionsClient sessionsClient = SessionsClient.create(sessionsSettings)) {
+      // Set the session name using the projectID (my-project-id), locationID (global), agentID
+      // (UUID), and sessionId (UUID).
+      SessionName session =
+          SessionName.ofProjectLocationAgentSessionName(projectId, locationId, agentId, sessionId);
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentAudioInput.java:75]: I wonder if this comment shouldn't be either for `detectIntent` or `main`

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentDisableWebhook.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import com.google.cloud.dialogflow.cx.v3.SessionName;
+import com.google.cloud.dialogflow.cx.v3.SessionsClient;
+import com.google.cloud.dialogflow.cx.v3.SessionsSettings;
+import com.google.cloud.dialogflow.cx.v3.TextInput;
+import com.google.common.collect.Maps;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class DetectIntentDisableWebhook {
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentDisableWebhook.java:36]: Should there be a `main()`?

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentDisableWebhook.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    } else {
+      sessionsSettingsBuilder.setEndpoint(locationId + "-dialogflow.googleapis.com:443");
+    }
+    SessionsSettings sessionsSettings = sessionsSettingsBuilder.build();
+
+    Map<String, QueryResult> queryResults = Maps.newHashMap();
+    // Instantiates a client
+    try (SessionsClient sessionsClient = SessionsClient.create(sessionsSettings)) {
+      // Set the session name using the projectID (my-project-id), locationID (global), agentID
+      // (UUID), and sessionId (UUID).
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentDisableWebhook.java:59]: perhaps part of this comment should be a method comment?

File: samples/snippets/src/main/java/dialogflow/cx/DetectIntentEventInput.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+public class DetectIntentEventInput {
+
+  // DialogFlow API Detect Intent sample with Event input.
+  public static void main(String[] args) throws IOException, ApiException {
+    String projectId = "my-project-id";
+    String locationId = "my-location-id";
+    String agentId = "my-agent-id";
+    String sessionId = "my-session-id";
+    String event = "my-event-id";
+    String languageCode = "my-language-code";
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/DetectIntentEventInput.java:41]: ditto

File: samples/snippets/src/test/java/dialogflow/cx/DetectIntentIntentInputTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+/** Unit test for {@link DetectIntentIntentInput}. */
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class DetectIntentIntentInputTest {
+
+  private static String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static String LOCATION = "global";
+  private static String AGENT_ID =
+      System.getenv()
+          .getOrDefault("DIALOGFLOW_CX_AGENT_ID_GLOBAL", "b8d0e85d-0741-4e6d-a66a-3671184b7b93");
+  private static String INTENT_ID = "00000000-0000-0000-0000-000000000000";
```
Comment (lesv) [inline samples/snippets/src/test/java/dialogflow/cx/DetectIntentIntentInputTest.java:38]: Should these be real UUID's?  Remember you might have 9 of these running at the same time in different processes / processors.

File: samples/snippets/src/test/java/dialogflow/cx/DetectIntentSentimentAnalysisTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+/** Unit test for {@link DetectIntentSentimentAnalysis}. */
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class DetectIntentSentimentAnalysisTest {
+
+  private static String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static String LOCATION = "global";
+  private static String AGENT_ID =
+      System.getenv()
+          .getOrDefault("DIALOGFLOW_CX_AGENT_ID_GLOBAL", "b8d0e85d-0741-4e6d-a66a-3671184b7b93");
```
Comment (lesv) [inline samples/snippets/src/test/java/dialogflow/cx/DetectIntentSentimentAnalysisTest.java:36]: Real UUID?

**Review by lesv — APPROVED** [review]
A few suggestions / questions:
* where UUID's are expected, perhaps you should actually create UUID's.
* Method comments may be appropriate in places.
* Tests don't appear to understand that they can be running in 9 separate instances at once.
* I don't see a lot of cleaning up of resources in tests.  Both what your creating in the moment, and stuff that might be left over after a crash.

File: samples/snippets/src/test/java/dialogflow/cx/DetectIntentIntentInputTest.java (hunk 8 above)
Comment (aribray) [reply samples/snippets/src/test/java/dialogflow/cx/DetectIntentIntentInputTest.java:38]: Each agent has a unique Agent ID that's created by Dialogflow. There was a permanent "TestAgent" resource created for the samples, and it has its own Flow IDs, Page IDs, etc. that are also created by Dialogflow (`00000000-0000-0000-0000-000000000000` is the Default Welcome Intent Flow ID).

Am I understanding the question correctly? I can go in and create a new intent by hand as well, if that makes more sense.

**Pull request merged**